plugins/in_dbus/in_dbus.c

```diff
@@ -0,0 +1,527 @@
+            flb_error("[in_dbus] Expected dictionary key");
+            break;
+        }
+
+        dbus_message_iter_recurse(&dict, &entry);
+        if (DBUS_TYPE_STRING != dbus_message_iter_get_arg_type(&entry)) {
+            flb_error("[in_dbus] Expected string key");
+            break;
+        }
+        dbus_message_iter_get_basic(&entry, &key);
+
+        dbus_message_iter_next(&entry);
+        if (DBUS_TYPE_VARIANT != dbus_message_iter_get_arg_type(&entry)) {
+            flb_error("[in_dbus] Expected variant key");
+            break;
+        }
+
+        dbus_message_iter_recurse(&entry, &variant);
+
+        {   /* Pack the key */
+            size_t key_len = strlen(key);
+            msgpack_pack_str(&dbus_config->mp_pck, key_len);
+            msgpack_pack_str_body(&dbus_config->mp_pck, key, key_len);
+        }
+
+        switch (dbus_message_iter_get_arg_type(&variant)) {
+#define GET(DBUS_TYPE, TYPE, MSGPACK_FUNC)                  \
+            case DBUS_TYPE: {                               \
+                TYPE v;                                     \
+                dbus_message_iter_get_basic(&variant, &v);  \
+                MSGPACK_FUNC(&dbus_config->mp_pck, v);      \
+                break;                                      \
+            }
+            GET(DBUS_TYPE_INT16,  int16_t,  msgpack_pack_int16);
+            GET(DBUS_TYPE_UINT16, uint16_t, msgpack_pack_uint16);
+            GET(DBUS_TYPE_INT32,  int32_t,  msgpack_pack_int32);
+            GET(DBUS_TYPE_UINT32, uint32_t, msgpack_pack_uint32);
+            GET(DBUS_TYPE_INT64,  int64_t,  msgpack_pack_int64);
+            GET(DBUS_TYPE_UINT64, uint64_t, msgpack_pack_uint64);
+            GET(DBUS_TYPE_DOUBLE, double,   msgpack_pack_double);
+            GET(DBUS_TYPE_BYTE,   uint8_t,  msgpack_pack_uint8);
+#undef GET
+            case DBUS_TYPE_STRING: {
+                char* s;
+                size_t len;
+
+                dbus_message_iter_get_basic(&variant, &s);
+                len = strlen(s);
+                msgpack_pack_str(&dbus_config->mp_pck, len);
+                msgpack_pack_str_body(&dbus_config->mp_pck, s, len);
+                break;
+            }
+            case DBUS_TYPE_BOOLEAN: {
+                dbus_bool_t b;
+                dbus_message_iter_get_basic(&variant, &b);
+                if (b) {
+                    msgpack_pack_true(&dbus_config->mp_pck);
+                }
+                else {
+                    msgpack_pack_false(&dbus_config->mp_pck);
+                }
+                break;
+            }
+            default:
+                flb_error("[in_dbus] Unknown type '%c'",
+                          dbus_message_iter_get_arg_type(&variant));
+                break;
+        }
+    }
+
+    /*  Send a DBus reply */
+    reply = dbus_message_new_method_return(msg);
+    if (!dbus_connection_send(conn, reply, &serial)) {
+        flb_error("[in_dbus] Could not send reply");
+    } else {
```

Comment:
The style guide forbids the [cuddled `else`](https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions).